### PR TITLE
builder: use DecomposeTransformedComponents filter

### LINKED ISSUE #367: Builder: use DecomposeTransformedComponents filter
Transformed components are problematic in variable fonts and with hinting.
Gftools builder should probably use the ufo2ft DecomposeTransformedComponents filter by default.

### PR DESCRIPTION
**Pull request by moyogo**
Fixes #367.

### CODE REVIEW

File: Lib/gftools/builder/__init__.py [hunk 1]
```diff
@@ -281,7 +285,22 @@ def run_fontmake(self, source, args):
...
+
+        if (
+            self.config["flattenComponents"] or
+            self.config["decomposeTransformedComponents"]
+        ):
+            filters = args.get("filters", [])
+            if self.config["flattenComponents"]:
+                filters.append([
+                    loadFilterFromString("FlattenComponentsFilter")
+                ])
```
Comment (m4rc1e) [inline Lib/gftools/builder/__init__.py:297]:
```suggestion
            if self.config["flattenComponents"]:
                filters.append(
                    loadFilterFromString("FlattenComponentsFilter")
                )
```

You're appending a list which contains a single filter. What you want is to just append the filter.

File: Lib/gftools/builder/__init__.py [hunk 2]
```diff
@@ -281,7 +285,22 @@ def run_fontmake(self, source, args):
...
+
+            if self.config["decomposeTransformedComponents"]:
+                filters.append([
+                    loadFilterFromString("DecomposeTransformedComponentsFilter")
+                ])
```
Comment (m4rc1e) [inline Lib/gftools/builder/__init__.py:302]:
```suggestion
                filters.append(
                    loadFilterFromString("DecomposeTransformedComponentsFilter")
                )
```

Same here :-)

Comment (m4rc1e) [reply Lib/gftools/builder/__init__.py:302]: Currently, ufo2ft doesn't like the lists being appended. If I remove them, then it's fine.

Comment (m4rc1e) [reply Lib/gftools/builder/__init__.py:302]: Here's a minimal test case

[component_test.zip](https://github.com/googlefonts/gftools/files/6584956/component_test.zip)

Comment (moyogo) [reply Lib/gftools/builder/__init__.py]: Thanks! It's fixed.

**Review by m4rc1e — APPROVED** [review]
LGTM. Thanks!

**Comment by m4rc1e** [comment]
I'll fix the CI in another pr.

**Pull request merged**